viewing_party/party.py

```diff
@@ -1,23 +1,102 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+    else:
+        return None
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:

```

Comment:
Note, while this may work, it's considered dangerous practice to modify the content of a data structure that you're looping over.  Especially in more complex code, this can open you up to complex errors. A simple alternative here could be to save a copy of what you get from movie in user_data["watchlist"] and iterate over that, so that as you modify user_data["watchlist"], there's no risk.

Reply:
Thank you, this is helpful. I was trying to avoid using more memory but did not realize the consequences as you described. Will be more careful in future.